src/lakefs_spec/spec.py

```diff
@@ -244,6 +344,24 @@ def info(self, path: str, **kwargs: Any) -> dict[str, Any]:
         }
 
     def ls(self, path: str, detail: bool = True, **kwargs: Any) -> list:
+        """
+        List all available objects under a given path in lakeFS.
+
+        Parameters
+        ----------
+        path: str
+            The path under which to list objects. Must be a fully qualified lakeFS URI.
+            Can also point to a file, in which case the file's metadata will be returned.
+        detail: bool
+            Whether to obtain all metadata on the requested objects or just their names.
+        kwargs: Any
+            Additional keyword arguments to pass to ``LakeFSClient.objects_api.list_objects()``.
+
+        Returns
+        -------
+        list[str | dict[str, Any]]
```

Comment:
Inconsistent with the method signature (but we can tighten the type bounds on the method since it's covariant with the base method)